### PR TITLE
update velocypack

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Update ArangoDB version of VelocyPack to latest.
This includes the following changes:
* copyright notice comments changed in all veloypack files
* improved APIs for `Slice::get()` and `Slice::hasKey()`, whose array variants now support std::initializer_list and any other forward-iterator-capable containers
* added HashedStringRef
* usage of `std::char_traits<T>::length(attribute)` for constexpr attribute names

- [ ] :hankey: Bugfix 
- [x] :pizza: New feature 
- [x] :hammer: Refactoring 
- [ ] :book: CHANGELOG entry made
- [ ] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

### Testing & Verification

- [x] There are tests in an external testing repository: arangodb/velocypack

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12310/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12312/

File: 3rdParty/velocypack/include/velocypack/HashedStringRef.h [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    : _data(data), 
+      _length(static_cast<uint32_t>(length)), 
+      _hash(hash(_data, _length)) {}
+ 
+  /// @brief create a HashedStringRef from a string plus length and hash.
+  /// note: the hash is *not* validated. it is the caller's responsibility
+  /// to ensure the hash value is correct
+  HashedStringRef(char const* data, std::size_t length, uint32_t hash) noexcept 
+    : _data(data), 
+      _length(static_cast<uint32_t>(length)),
```
Comment (gnusi) [inline 3rdParty/velocypack/include/velocypack/HashedStringRef.h:64]: do you think it's safe enough to silently cast `size_t` to `uint32_t`?

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12317/

File: 3rdParty/velocypack/include/velocypack/HashedStringRef.h (hunk 1 above)
Comment (jsteemann) [reply 3rdParty/velocypack/include/velocypack/HashedStringRef.h]: addressed

**Review by gnusi — APPROVED** [review]

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12318/

**Pull request merged**